### ISSUE TITLE
httpd-init doesn't deploy on openshift when using cri-o runtime

### ISSUE THREAD

**Issue by carbonin** [ManageIQ/manageiq-pods#361]
Currently it seems CRI-O and the oci-systemd-hooks used for the httpd pod don't seem to play nicely together, but we should be able to run on openshift systems using cri-o.

Some more details are in https://bugzilla.redhat.com/show_bug.cgi?id=1652457

-------------------------------------------------------------

Summary:
The oci-systemd-hooks were removed from OpenShift in https://github.com/openshift/machine-config-operator/pull/1314 this prevents us from running httpd with `init` as pid 1 without privileged.

I opened https://bugzilla.redhat.com/show_bug.cgi?id=1807245 against OpenShift to address the issue.

**Comment by carbonin** [comment]
This seems to be a privileges issue. Things work when we deploy httpd using the privileged scc.

Will have to do some more digging to figure out what the difference is and what the "correct" way to accomplish what we're trying to do would be.

I also found https://bugzilla.redhat.com/show_bug.cgi?id=1777838#c11 which mentions oci-systemd-hooks possibly being deprecated.

**Comment by carbonin** [comment]
@Fryguy

**Comment by carbonin** [comment]
I hope https://github.com/ManageIQ/container-httpd/pull/44 may help resolve this, but I don't have the environment tp test it.

@bennett-white @sethpeterson would you guys be able to help out as you were able to reproduce this issue in your environment. Specifically I think this should now work with the any-uid scc rather than privileged.

**Comment by carbonin** [comment]
Built out a [repo](https://github.com/carbonin/crio-systemd-test) to help test this in isolation. In addition to standalone deployments and buildconfigs it has a Dockerfile which builds a very simple httpd container on top of ubi-init to use as a baseline.

I've also been using [Code Ready Containers](https://github.com/code-ready/crc) to test this locally on openshift 4.

So far neither container works for me on openshift 4.3 cri-o version `cri-o://1.16.2-6.dev.rhaos4.3.git9e3db66.el8` and both containers worked for me in openshift 3.11 cri-o version `cri-o://1.11.16-0.5.dev.rhaos3.11.git3f89eba.el7-dev`

**Comment by carbonin** [comment]
Following the steps in https://github.com/carbonin/crio-systemd-test/blob/master/README.md using crc version `1.5.0+e1c8fb8` which includes openshift `4.2.14` I was able to get both containers up and running with the anyuid scc.

This is using cri-o version `cri-o://1.14.11-4.dev.rhaos4.2.git179ea6b.el8`. Going to re-test with the newer crc release to be sure it doesn't work there.

**Comment by carbonin** [comment]
Confirmed the failure in the following environment:
```
16:27:11:~/Source/crio-systemd-test (master)$ crc version
crc version: 1.6.0+8ef676f
OpenShift version: 4.3.0 (embedded in binary)
16:27:23:~/Source/crio-systemd-test (master)$ oc get nodes -owide
NAME                 STATUS    ROLES           AGE       VERSION   INTERNAL-IP      EXTERNAL-IP   OS-IMAGE                                                       KERNEL-VERSION                CONTAINER-RUNTIME
crc-w6th5-master-0   Ready     master,worker   11d       v1.16.2   192.168.130.11   <none>        Red Hat Enterprise Linux CoreOS 43.81.202001142154.0 (Ootpa)   4.18.0-147.3.1.el8_1.x86_64   cri-o://1.16.2-6.dev.rhaos4.3.git9e3db66.el8
```

Going to write up an openshift bug

**Comment by carbonin** [comment]
Opened https://bugzilla.redhat.com/show_bug.cgi?id=1807245 against crc as that's where I reproduced the issue. Assuming it will get moved to some other component.

**Comment by carbonin** [comment]
[Commented](https://github.com/openshift/machine-config-operator/pull/1314#issuecomment-605166089) on the PR that seems to have caused the issue (at least the most recent version of it).

**Comment by carbonin** [comment]
PR has been opened on cri-o to address the issue https://github.com/cri-o/cri-o/pull/3506

**Comment by carbonin** [comment]
So the "fix" for this was released in openshift 4.3.13, but really we'll still need to set the `container_manage_cgroup` on nodes to make this work correctly.

I raised this concern in https://bugzilla.redhat.com/show_bug.cgi?id=1821511#c10, but haven't gotten a response in the last week.

**Comment by gyliu513** [comment]
@carbonin how to handle the case if customer using OCP below 4.3.13?

**Comment by carbonin** [comment]
Well like I said before, this works fine in 3.11, but I'm not sure about 4.1 and 4.2

**Comment by carbonin** [comment]
According to https://access.redhat.com/support/policy/updates/openshift both 4.1 and 4.2 are in "Maintenance Support" so I don't think we should expect this to come back as it's not a "Urgent and Selected High Priority Bug Fix"

**Comment by gyliu513** [comment]
> it's not a "Urgent and Selected High Priority Bug Fix"

The manageIQ operator does not work and I cannot access the UI in this case, curious why this is not a High Priority Bug :-)

**Comment by carbonin** [comment]
As much as I would like them to be, the needs of ManageIQ are not at the top of the priority list for the OpenShift team.

**Comment by carbonin** [comment]
Looks like this will actually be fixed by https://github.com/cri-o/cri-o/pull/3754 which it seems will be released in crio-1.18.

According to https://bugzilla.redhat.com/show_bug.cgi?id=1807245#c14 it sounds like this will be available in OpenShift 4.5 and 4.4, not sure about earlier versions.

**Comment by carbonin** [comment]
The bug was finally verified, so I'm going to close this here as we're very much less reliant on running the httpd-init image and we can revisit this with a new issue if someone hits this again on a newer version of OpenShift. (I think >= 4.5 will have the fix).
